File: app/src/main/java/fr/legrand/oss117soundboard/data/manager/db/DatabaseManager.java
package fr.legrand.oss117soundboard.data.manager.db;

import com.raizlabs.android.dbflow.annotation.Database;

import java.util.List;

import fr.legrand.oss117soundboard.data.entity.Reply;
import io.reactivex.Observable;

/**
 * Created by Benjamin on 30/09/2017.
 */
@Database(name = DatabaseManager.NAME, version = DatabaseManager.VERSION)
public interface DatabaseManager {
    String NAME = "OSSDatabase";
    int VERSION = 1;

    List<Reply> getReplyWithSearch(String search, boolean fromFavorite);
    List<Reply> getAllReply(boolean fromFavorite);
    List<Reply> getAllReply();

    Reply updateFavoriteReply(int replyId, boolean addToFavorite);

    void saveReplyList(List<Reply> replyList);

    Reply getMostListenedReply();

    void incrementReplyListenCount(int replyId);
}

File: app/src/main/java/fr/legrand/oss117soundboard/data/manager/db/DatabaseManager.kt
package fr.legrand.oss117soundboard.data.manager.db

import com.raizlabs.android.dbflow.annotation.Database
import fr.legrand.oss117soundboard.data.entity.Reply

/**
 * Created by Benjamin on 30/09/2017.
 */
@Database(name = DatabaseManagerImpl.NAME, version = DatabaseManagerImpl.VERSION)
interface DatabaseManager {
    fun getAllReply(): List<Reply>

    fun getMostListenedReply(): Reply

    fun getReplyWithSearch(search: String, fromFavorite: Boolean): List<Reply>
    fun getAllReply(fromFavorite: Boolean): List<Reply>

    fun updateFavoriteReply(replyId: Int, addToFavorite: Boolean): Reply

    fun saveReplyList(replyList: List<Reply>)

    fun incrementReplyListenCount(replyId: Int)


}
